File: lib/src/main/java/co/infinum/goldfinger/LegacyGoldfinger.java
package co.infinum.goldfinger;

/**
 * Legacy implementation for pre-Marshmallow devices.
 */
class LegacyGoldfinger implements Goldfinger {

    @Override
    public boolean hasFingerprintHardware() {
        return false;
    }

    @Override
    public boolean hasEnrolledFingerprint() {
        return false;
    }

    @Override
    public void authenticate(Callback callback) {
    }

    @Override
    public void decrypt(String keyName, String value, Callback callback) {
    }

    @Override
    public void encrypt(String keyName, String value, Callback callback) {
    }

    @Override
    public void cancel() {
    }
}

File: lib/src/main/kotlin/co.infinum.goldfinger/LegacyGoldfinger.kt
package co.infinum.goldfinger

/**
 * Legacy implementation for pre-Marshmallow devices.
 */
internal class LegacyGoldfinger : Goldfinger {
    override fun authenticate(callback: Goldfinger.Callback) {}
    override fun decrypt(keyName: String, value: String, callback: Goldfinger.Callback) {}
    override fun encrypt(keyName: String, value: String, callback: Goldfinger.Callback) {}
    override fun hasFingerprintHardware() = false
    override fun hasEnabledLockScreen() = false
    override fun hasEnrolledFingerprint() = false
    override fun cancel() {}
}